src/Functions/seriesPeriodDetectFFT.cpp

```diff
@@ -153,12 +153,8 @@ class FunctionSeriesPeriodDetectFFT : public IFunction
             return true;
         }
 
-        std::vector<double> xfreq(spec_len);
         double step = 0.5 / (spec_len - 1);
-        for (size_t i = 0; i < spec_len; ++i)
-            xfreq[i] = i * step;
-
-        auto freq = xfreq[idx];
```

Comment:
`xfreq` is calculated for every frequency but only the `idx` one is used. This patch removes other unneeded calculations and use `idx * step` directly.

Reply:
very strange code, indeed